addons/stock/models/stock_picking.py

```diff
@@ -693,9 +699,13 @@ def action_confirm(self):
         self._check_company()
         self.mapped('package_level_ids').filtered(lambda pl: pl.state == 'draft' and not pl.move_ids)._generate_moves()
         # call `_action_confirm` on every draft move
-        self.mapped('move_lines')\
-            .filtered(lambda move: move.state == 'draft')\
-            ._action_confirm()
+        moves = self.mapped('move_lines')\
+                    .filtered(lambda move: move.state == 'draft')\
+                    ._action_confirm()
+        moves.filtered(lambda m: m.picking_id.picking_type_id.reservation_method == 'at_confirm'
+                                 or (m.picking_id.picking_type_id.reservation_method == 'by_date'
+                                    and m.reservation_date <= fields.Datetime.now()))\
+             ._action_assign()
```

Comment:
I would rather complete this part https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L1125

Reply:
I did a first cut at this in the fixup, but for some reason its causing a weird package move usererror when the demand isn't fully completed (i.e. the currently failing test) and the products are stored in packages (but not going into packages). Will debug more next week...